Feat/create custom hook for use products

imattking opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and agree to adhere to the [Contribution Guidelines](https://github.com/freeCodeCamp-2025-Summer-Hackathon/purple-array/blob/main/contribution-guidelines.md).
- [x] My pull request targets the `main` branch of the repository.
- [x] I have tested these changes either locally on my machine, or Gitpod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #96 

<!-- Feel free to add any additional description of changes below this line -->
### This PR covers the following changes:
#### on the backend:
- adds Cors package to backend
- sets up use of Cors as middleware on our server when running in dev mode
#### on the frontend: 
- fixes an issue with lib/axios.js expecting a PORT variable when it should not have
- adds the util/api folder
  - makes use of the axios util file define necessary fetch request on the front end
- adds the util/hooks folder
  -  houses custom React hooks to import util/api files and employ async data & error handling before passing that data along to front end components <br>
|--src
| ------ |___ util
| ------------ |___ api
| ------------ |___ hooks

 ***the above structure will keep components free of housing large blocks of state declarations and useEffect calls***

### files added in this PR:
- **products.js** — /frontend/src/util/api/products.js 
- **useProducts.js** — /frontend/src/util/hooks/useProducts.js

haednodik reviewed (changes requested):
Looks good to me and I can see the products. There is a .DS_Store that shouldn't be in the commit. I'm not sure if .DS_Store? in the .gitignore is working or should just be .DS_Store.

imattking commented:
Got it. I will check and see if I can tell where it's coming from, and I'll double check the .gitignore to make sure it's listed there as well!

imattking commented:
@haednodik I've removed the .DS_store file, committed and pushed that change, updated the .gitignore in another PR, and resolved the merge conflicts on this branch that cropped up (in the server.js, package.json and package-lock.json) between now and when this PR was created!

Let me know what else you need!

haednodik reviewed (approved):
All good
